test/classes/test_combinatorial_complex.py

```diff
@@ -67,8 +67,13 @@ def test_add_cell(self):
         """Test adding a cell to a CC."""
         CC = CombinatorialComplex()
         CC.add_cell([1, 2, 3], rank=2)
-
         assert (1, 2, 3) in CC.cells
+        with pytest.raises(ValueError) as ex:
+            CC.add_cell(HyperEdge([1]), 3)
+            assert (
+                str(ex.value)
+                == "violation of the combinatorial complex condition : the hyperedge frozenset({1, 2, 3}) in the complex has rank 2 is smaller than 3, the rank of the input hyperedge frozenset({1}) "
+            )
```

Comment:
This assert statement shouldn't be within the pytest context. 
```suggestion
        assert (
                str(ex.value)
                == "violation of the combinatorial complex condition : the hyperedge frozenset({1, 2, 3}) in the complex has rank 2 is smaller than 3, the rank of the input hyperedge frozenset({1}) "
            )
```